src/image.tsx

```diff
@@ -143,23 +160,24 @@ const Image = React.forwardRef<HTMLImageElement, ImageProps>(
                     })
                 )}
                 alt={alt}
-                onError={(event) => {
+                onError={() => {
                     setIsError(true);
                     setIsLoading(false);
                     setHideLoadingFallback(true);
-                    onError?.(event);
-                }}
-                onLoad={(event) => {
-                    setIsError(false);
-                    setIsLoading(false);
-                    setTimeout(() => {
-                        setHideLoadingFallback(true);
-                    }, styles.FADE_IN_DURATION_MS);
-                    onLoad?.(event);
+                    onError?.();
                 }}
+                onLoad={onLoadHandler}
             />
         );
 
+        React.useEffect(() => {
+            // Needed because there is some race condition with SSR and onLoad events
+            // https://github.com/facebook/react/issues/15446
+            if (imageRef.current?.complete) {
+                onLoadHandler();
```

Comment:
This can be called multiple times if the father doesn't use `React.useCallback`... could that be a problem?